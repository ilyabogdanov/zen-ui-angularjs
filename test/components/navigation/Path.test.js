const div = document.createElement("div");
document.body.appendChild(div);

describe("directive: zen-path", function() {
    let element, scope;
    beforeEach(angular.mock.module("ZenUI"));
    beforeEach(inject(function($compile, $rootScope) {
        scope = $rootScope.$new();
        element = ""+
            "<zen-path>" +
            "<zen-path-item zen-href=\"::'/'\"></zen-path-item>" +
            "<zen-path-item zen-sref=\"::'___'\"></zen-path-item>" +
            "</zen-path>";
        element = $compile(element)(scope);
        scope.$digest();
        angular.element(div).append(element);
    }));
    afterEach(function () {
        while (div.firstChild) {
            div.removeChild(div.firstChild);
        }
    });

    it("should render", function () {
        expect(div.childNodes).toHaveLength(1);
        expect(div.firstChild).toMatchSnapshot();
    });
});
