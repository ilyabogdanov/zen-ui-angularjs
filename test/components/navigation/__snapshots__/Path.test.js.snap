// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`directive: zen-path should render 1`] = `
<div
  class="zen_ui__path ng-scope"
>
  <div>
    <div
      ng-transclude=""
    >
      <span
        class="zen_ui__path_item ng-scope ng-isolate-scope"
        zen-href="::'/'"
      >
        <a
          class="link ng-binding"
          href="unsafe:/"
          ng-bind="zenText"
          ng-href="/"
        />
      </span>
      <span
        class="zen_ui__path_item ng-scope ng-isolate-scope"
        zen-sref="::'___'"
      >
        <a
          class="link ng-binding"
          ng-bind="zenText"
          ui-sref="___"
        />
      </span>
    </div>
  </div>
</div>
`;
